File: app/extractors/DataExtractor.js
import { readdirSync, readFileSync } from 'fs';

const INPUT_FOLDER = './raw_data/';

function DataExtractor() {
  // For this sample, the extractor reads files from the local disk.
  // In practice, it would get data from an external system.

  this.extract = function extract() {
    const filenames = readdirSync(INPUT_FOLDER);
    return filenames.map((filename) => ({
      basename: filename.replace(/\..*/, ''),
      content: readFileSync(`${INPUT_FOLDER}/${filename}`, 'utf-8'),
    }));
  };
}

export default DataExtractor;

File: app/extractors/DataExtractor.py
from pathlib import Path
from typing import List
from ..data_types import ExtractedData

INPUT_FOLDER = Path("./raw_data/")


class DataExtractor:
    """Extracts data from raw data files.

    For this sample, the extractor reads files from the local disk.
    In practice, it would get data from an external system.
    """

    def extract(self) -> List[ExtractedData]:
        filenames = sorted(INPUT_FOLDER.iterdir())
        return [
            {
                "basename": filename.stem,
                "content": filename.read_text(encoding="utf-8"),
            }
            for filename in filenames
        ]
